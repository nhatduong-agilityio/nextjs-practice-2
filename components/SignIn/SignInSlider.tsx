'use client'

// Libs
import React, { memo } from 'react'
import { HStack, Heading, Stack } from '@chakra-ui/react'
import dynamic from 'next/dynamic'

// Components
import BagIcon from '@components/Icons/BagIcon'
import BrandIcon from '@components/Icons/BrandIcon'

const Slider = dynamic(() => import('@components/Slider/Slider'))

type TSlide = {
  id: string
  image: string
  imageAlt: string
  content: {
    title: string
    description: string
  }
}

interface SignInSliderProps {
  sliders: TSlide[]
}

const SignInSlider = memo(({ sliders }: SignInSliderProps) => {
  return (
    <Stack
      width={['full', '50%']}
      flexDir='column'
      gap={0}
      bgColor='blue.50'
      pt={['25px', '40px']}
      pl={['5', '100px']}
      pr={['5', '77px']}
    >
      <HStack alignItems='end'>
        <BagIcon width={['27px', '38px']} height={['27px', '38px']} fill='black' />
        <BrandIcon
          width={['63px', '88px']}
          height={['15px', '20px']}
          viewBox='0 0 63 15'
          fill='black'
        />
      </HStack>
      <Heading
        variant={['extraBoldMD', 'extraBoldXL']}
        pt={['5', '46px']}
        pb={['5', '50px']}
        lineHeight='normal'
      >
        Furniture Marketplace
      </Heading>
      <Slider
        slides={sliders}
        timerChange={3000}
        heightSlider={['200px', '547px']}
        sxControl={{ boxSize: ['8px', '15px'] }}
        variant='image'
      />
    </Stack>
  )
})

SignInSlider.displayName = 'SignInSlider'

export default SignInSlider
