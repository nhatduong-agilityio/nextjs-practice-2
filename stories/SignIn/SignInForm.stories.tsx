import SignInForm from '@components/SignIn/SignInForm'
import { INVALID_PASSWORD, USERNAME_NOT_EXIST } from '@constants/errorMessage'
import type { SignInAccount } from '@matched-types/auth'
import { UserMock } from '@mock/dataMock'
import type { Meta } from '@storybook/react'
import { findItemByValue } from '@utils/index'
import { signInValidate } from '@utils/signInValidate'
import { useCallback, useState } from 'react'
import type { SubmitHandler } from 'react-hook-form'

const meta: Meta<typeof SignInForm> = {
  title: 'SignIn/SignForm',
  component: SignInForm,
}

export default meta

const initErrorMsgs = {
  username: '',
  password: '',
}

const SignInFormStories = () => {
  const [errorMsgs, setErrorMsgs] = useState(initErrorMsgs)

  // Function submit signIn form
  const handleSubmit: SubmitHandler<SignInAccount> = useCallback(async (data) => {
    const formatValidation = signInValidate(data)

    if (!formatValidation.isValid && formatValidation.error) {
      setErrorMsgs(formatValidation.error)
    } else {
      const users = findItemByValue({
        data: UserMock,
        value: data.username,
        key: 'username',
      })

      if (!users) {
        setErrorMsgs((prev) => ({
          ...prev,
          username: USERNAME_NOT_EXIST,
        }))
      } else if (users && users.password !== data.password) {
        setErrorMsgs((prev) => ({
          ...prev,
          password: INVALID_PASSWORD,
        }))
      } else {
        // eslint-disable-next-line no-alert
        alert(`Login Successfully!`)

        setErrorMsgs(initErrorMsgs)
      }
    }
  }, [])

  return (
    <SignInForm
      usernameError={errorMsgs.username}
      passwordError={errorMsgs.password}
      onSubmit={handleSubmit}
    />
  )
}

export const Default = {
  render: () => <SignInFormStories />,
}
