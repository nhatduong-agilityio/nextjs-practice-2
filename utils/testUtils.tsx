import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
