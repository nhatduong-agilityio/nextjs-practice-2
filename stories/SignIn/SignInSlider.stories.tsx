import SignInSlider from '@components/SignIn/SignInSlider'
import { SlideMock } from '@mock/dataMock'
import type { Meta } from '@storybook/react'

const meta: Meta<typeof SignInSlider> = {
  title: 'SignIn/SignInSlider',
  component: SignInSlider,
}

export default meta

const SignInSliderStories = () => {
  return <SignInSlider sliders={SlideMock} />
}

export const Default = {
  render: () => <SignInSliderStories />,
}
