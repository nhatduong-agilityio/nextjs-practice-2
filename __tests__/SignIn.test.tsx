import { render, waitFor } from '@utils/testUtils'
import SignIn from '../app/(auth)/signin/page'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))

describe('SignIn render', () => {
  it('Should SignIn match Snapshot', async () => {
    const { container } = render(<SignIn />)

    let lazyContainer

    await waitFor(() => {
      lazyContainer = container
    })

    expect(lazyContainer).toMatchSnapshot()
  })
})
