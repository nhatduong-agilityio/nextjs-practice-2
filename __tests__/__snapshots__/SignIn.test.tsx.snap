// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn render Should SignIn match Snapshot 1`] = `
<div>
  <div
    class="css-8s5t6j"
  >
    <div
      class="chakra-stack css-1q6zlog"
    >
      <div
        class="chakra-stack css-uk2lwn"
      >
        <svg
          class="chakra-icon css-bu03u0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.7178 7.68579C22.7096 7.57546 22.6172 7.49011 22.5067 7.49011H18.6948V6.43125C18.6948 3.0448 15.9396 0.289856 12.5534 0.289856C9.16715 0.289856 6.412 3.0448 6.412 6.43125V7.49011H2.6001C2.48955 7.49011 2.39722 7.57546 2.38896 7.68579L0.906552 27.1688C0.902105 27.2275 0.922223 27.2855 0.96246 27.3287C1.00248 27.3721 1.05882 27.3967 1.11769 27.3967H23.9891C24.048 27.3967 24.1043 27.3721 24.1443 27.3289C24.1844 27.2857 24.2047 27.2277 24.2002 27.169L22.7178 7.68579ZM8.10617 6.43125C8.10617 3.97914 10.1011 1.98403 12.5534 1.98403C15.0057 1.98403 17.0006 3.97914 17.0006 6.43125V7.49011H8.10617V6.43125ZM7.25909 13.4197C6.09116 13.4197 5.14136 12.4697 5.14136 11.302C5.14136 10.4568 5.56491 9.6953 6.412 9.36112V11.302C6.412 11.7692 6.79213 12.1491 7.25909 12.1491C7.72604 12.1491 8.10617 11.7692 8.10617 11.302V9.36239C8.95326 9.69657 9.37681 10.4638 9.37681 11.302C9.37681 12.4697 8.42701 13.4197 7.25909 13.4197ZM17.8477 13.4197C16.6798 13.4197 15.73 12.4697 15.73 11.302C15.73 10.4568 16.1535 9.6953 17.0006 9.36112V11.302C17.0006 11.7692 17.3807 12.1491 17.8477 12.1491C18.3147 12.1491 18.6948 11.7692 18.6948 11.302V9.36239C19.5419 9.69657 19.9654 10.4638 19.9654 11.302C19.9654 12.4697 19.0156 13.4197 17.8477 13.4197Z"
          />
        </svg>
        <svg
          class="chakra-icon css-bj7ht2"
          focusable="false"
          viewBox="0 0 63 15"
        >
          <path
            d="M7.03922 8.17836C7.03922 8.77838 6.88701 9.31222 6.58259 9.77989C6.27817 10.2431 5.83918 10.6049 5.26563 10.8652C4.69208 11.1211 4.01926 11.2491 3.24718 11.2491C2.60304 11.2491 2.06258 11.2049 1.6258 11.1167C1.18902 11.024 0.734588 10.8652 0.262512 10.6402V8.31072C0.761059 8.56661 1.27946 8.76735 1.81771 8.91294C2.35597 9.05413 2.8501 9.12472 3.30012 9.12472C3.68837 9.12472 3.97294 9.05854 4.15383 8.92618C4.33472 8.78941 4.42516 8.61514 4.42516 8.40337C4.42516 8.27101 4.38766 8.1563 4.31266 8.05924C4.24207 7.95776 4.12515 7.85629 3.96191 7.75482C3.80308 7.65334 3.37512 7.44598 2.67804 7.13273C2.04713 6.84596 1.57285 6.56801 1.25519 6.29888C0.941948 6.02975 0.708116 5.72092 0.553699 5.37238C0.403694 5.02384 0.328691 4.61132 0.328691 4.13483C0.328691 3.24363 0.652967 2.54875 1.30152 2.0502C1.95007 1.55165 2.84128 1.30238 3.97514 1.30238C4.97665 1.30238 5.99801 1.53401 7.03922 1.99726L6.23846 4.01571C5.33402 3.60099 4.55311 3.39363 3.89573 3.39363C3.55601 3.39363 3.30894 3.45319 3.15453 3.57231C3.00011 3.69144 2.9229 3.83924 2.9229 4.01571C2.9229 4.20542 3.01996 4.37528 3.21409 4.52529C3.41262 4.67529 3.94647 4.94883 4.81562 5.34591C5.64947 5.72092 6.22743 6.12461 6.5495 6.55698C6.87598 6.98494 7.03922 7.5254 7.03922 8.17836Z"
          />
          <path
            d="M13.1012 11.1167V7.02685C13.1012 6.02093 12.8034 5.51797 12.2078 5.51797C11.7842 5.51797 11.471 5.69665 11.2681 6.05402C11.0695 6.40697 10.9702 6.99597 10.9702 7.82099V11.1167H8.38266V0.819275H10.9702V2.28183C10.9702 2.94362 10.935 3.69805 10.8644 4.54514H10.9835C11.2306 4.15689 11.5239 3.88335 11.8637 3.72453C12.2034 3.5657 12.5982 3.48628 13.0483 3.48628C13.8865 3.48628 14.5373 3.72894 15.0005 4.21425C15.4682 4.69515 15.702 5.37017 15.702 6.23932V11.1167H13.1012Z"
          />
          <path
            d="M24.6229 7.35112C24.6229 8.57764 24.292 9.53503 23.6303 10.2233C22.9729 10.9071 22.0508 11.2491 20.864 11.2491C19.7257 11.2491 18.8191 10.8983 18.144 10.1968C17.4734 9.49532 17.1381 8.54676 17.1381 7.35112C17.1381 6.12902 17.4668 5.18046 18.1242 4.50544C18.786 3.826 19.7125 3.48628 20.9037 3.48628C21.6405 3.48628 22.2912 3.6429 22.856 3.95615C23.4207 4.2694 23.8553 4.71941 24.1597 5.3062C24.4685 5.88857 24.6229 6.57021 24.6229 7.35112ZM19.7654 7.35112C19.7654 7.99527 19.8514 8.4872 20.0235 8.82691C20.1956 9.16222 20.4846 9.32987 20.8905 9.32987C21.2919 9.32987 21.5743 9.16222 21.7375 8.82691C21.9052 8.4872 21.989 7.99527 21.989 7.35112C21.989 6.7114 21.9052 6.22829 21.7375 5.90181C21.5699 5.57533 21.2831 5.41209 20.8772 5.41209C20.4801 5.41209 20.1956 5.57533 20.0235 5.90181C19.8514 6.22388 19.7654 6.70698 19.7654 7.35112Z"
          />
          <path
            d="M30.6055 11.2491C30.2261 11.2491 29.8864 11.1873 29.5864 11.0638C29.2863 10.9358 28.9885 10.7086 28.6929 10.3821H28.5738C28.6532 10.9071 28.6929 11.2336 28.6929 11.3616V14.3727H26.1053V3.61864H28.2098L28.5738 4.57161H28.6929C29.1738 3.84806 29.8378 3.48628 30.6849 3.48628C31.5144 3.48628 32.1651 3.83041 32.6372 4.51867C33.1093 5.20693 33.3453 6.15108 33.3453 7.35112C33.3453 8.55999 33.0982 9.51297 32.6041 10.2101C32.1144 10.9027 31.4482 11.2491 30.6055 11.2491ZM29.7386 5.51797C29.3591 5.51797 29.0922 5.65474 28.9378 5.92828C28.7878 6.19741 28.7062 6.59889 28.6929 7.13273V7.33789C28.6929 7.98203 28.779 8.44749 28.951 8.73426C29.1231 9.02104 29.3944 9.16442 29.765 9.16442C30.0959 9.16442 30.3364 9.02324 30.4864 8.74088C30.6364 8.45411 30.7114 7.98203 30.7114 7.32465C30.7114 6.68051 30.6364 6.21947 30.4864 5.94152C30.3364 5.65915 30.0871 5.51797 29.7386 5.51797Z"
          />
          <path
            d="M38.5006 1.44136H41.8758C43.1905 1.44136 44.1656 1.63769 44.8009 2.03035C45.4406 2.4186 45.7605 3.02524 45.7605 3.85027C45.7605 4.39293 45.6215 4.85618 45.3435 5.24002C45.0656 5.62386 44.6994 5.87093 44.245 5.98122V6.0474C44.845 6.20623 45.2796 6.47095 45.5487 6.84155C45.8178 7.21215 45.9524 7.69305 45.9524 8.28425C45.9524 9.16222 45.6215 9.85489 44.9597 10.3623C44.3023 10.8652 43.4045 11.1167 42.2662 11.1167H38.5006V1.44136ZM41.1147 5.15399H41.9022C42.2772 5.15399 42.5662 5.07678 42.7692 4.92236C42.9765 4.76794 43.0802 4.53852 43.0802 4.2341C43.0802 3.69144 42.6699 3.4201 41.8493 3.4201H41.1147V5.15399ZM41.1147 7.06656V9.09825H42.0346C42.8508 9.09825 43.2589 8.75412 43.2589 8.06586C43.2589 7.74379 43.1486 7.49672 42.928 7.32465C42.7118 7.15259 42.3964 7.06656 41.9816 7.06656H41.1147Z"
          />
          <path
            d="M52.3717 11.1167L51.8754 10.124H51.8224C51.4739 10.5564 51.1187 10.852 50.757 11.0108C50.3952 11.1696 49.9275 11.2491 49.354 11.2491C48.6481 11.2491 48.0922 11.0373 47.6863 10.6137C47.2804 10.1902 47.0774 9.59459 47.0774 8.82691C47.0774 8.02836 47.3554 7.43495 47.9113 7.0467C48.4672 6.65404 49.2724 6.43345 50.3268 6.38491L51.5776 6.34521V6.23932C51.5776 5.62165 51.2732 5.31282 50.6643 5.31282C50.1172 5.31282 49.4224 5.49812 48.5797 5.86872L47.8319 4.16131C48.7054 3.71129 49.8084 3.48628 51.1408 3.48628C52.1026 3.48628 52.846 3.72453 53.371 4.20101C53.9005 4.6775 54.1652 5.3437 54.1652 6.19961V11.1167H52.3717ZM50.4525 9.42252C50.7658 9.42252 51.0327 9.32325 51.2533 9.12472C51.4783 8.92618 51.5908 8.66808 51.5908 8.35043V7.76805L50.9952 7.79452C50.1437 7.82541 49.718 8.13865 49.718 8.73426C49.718 9.1931 49.9628 9.42252 50.4525 9.42252Z"
          />
          <path
            d="M63 3.61864V4.86942L61.9743 5.24664C62.1155 5.50253 62.186 5.80033 62.186 6.14005C62.186 6.94743 61.9015 7.57393 61.3323 8.01953C60.7676 8.46514 59.9227 8.68794 58.7977 8.68794C58.5197 8.68794 58.2991 8.67029 58.1359 8.63499C58.0741 8.7497 58.0433 8.8578 58.0433 8.95927C58.0433 9.08722 58.1469 9.18648 58.3543 9.25708C58.5616 9.32325 58.833 9.35634 59.1683 9.35634H60.4125C62.0934 9.35634 62.9339 10.0645 62.9339 11.4807C62.9339 12.394 62.5434 13.1043 61.7625 13.6116C60.986 14.119 59.8941 14.3727 58.4866 14.3727C57.4234 14.3727 56.6028 14.194 56.0248 13.8366C55.4512 13.4837 55.1645 12.9741 55.1645 12.3079C55.1645 11.4123 55.7204 10.8343 56.8322 10.574C56.6028 10.477 56.402 10.3203 56.2299 10.1042C56.0623 9.88798 55.9785 9.674 55.9785 9.46223C55.9785 9.2284 56.0424 9.02324 56.1704 8.84677C56.2983 8.66588 56.5653 8.4519 56.9712 8.20483C56.5829 8.03277 56.2763 7.76364 56.0513 7.39745C55.8307 7.03126 55.7204 6.58566 55.7204 6.06064C55.7204 5.24443 55.9983 4.61132 56.5542 4.16131C57.1101 3.71129 57.9043 3.48628 58.9367 3.48628C59.0734 3.48628 59.3095 3.50172 59.6448 3.53261C59.9801 3.56349 60.2271 3.59217 60.386 3.61864H63ZM57.3484 12.0498C57.3484 12.2616 57.4631 12.4315 57.6925 12.5594C57.9219 12.6918 58.2285 12.7579 58.6124 12.7579C59.2389 12.7579 59.7396 12.6785 60.1146 12.5197C60.4941 12.3653 60.6838 12.1645 60.6838 11.9175C60.6838 11.7189 60.5691 11.5777 60.3397 11.4939C60.1102 11.4145 59.7815 11.3748 59.3536 11.3748H58.3411C58.0631 11.3748 57.8271 11.4388 57.6329 11.5667C57.4432 11.6991 57.3484 11.8601 57.3484 12.0498ZM58.1624 6.07387C58.1624 6.84155 58.4293 7.22539 58.9631 7.22539C59.2102 7.22539 59.3999 7.13053 59.5323 6.94082C59.669 6.7511 59.7374 6.46653 59.7374 6.08711C59.7374 5.31061 59.4793 4.92236 58.9631 4.92236C58.4293 4.92236 58.1624 5.3062 58.1624 6.07387Z"
          />
        </svg>
      </div>
      <h2
        class="chakra-heading css-zgeo2e"
      >
        Furniture Marketplace
      </h2>
      <div
        class="css-1ueoeyi"
        style="margin-top: 0px;"
      >
        <div
          class="css-6u8kiw"
        >
          <div
            class="css-kut80o"
          >
            <div
              class="css-ujemdd"
            >
              <div
                class="css-1gfsnss"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Picture of furniture chair leather"
                    data-nimg="fill"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
            </div>
            <div
              class="css-ujemdd"
            >
              <div
                class="css-1gfsnss"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Picture of furniture desk office"
                    data-nimg="fill"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
            </div>
            <div
              class="css-ujemdd"
            >
              <div
                class="css-1gfsnss"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Picture of furniture sofa leather"
                    data-nimg="fill"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
            </div>
            <div
              class="css-ujemdd"
            >
              <div
                class="css-1gfsnss"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Picture of furniture chair office"
                    data-nimg="fill"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-jfkb22"
        >
          <div
            class="css-1sxs0wg"
          />
          <div
            class="css-19iw3qs"
          />
          <div
            class="css-1sxs0wg"
          />
          <div
            class="css-1sxs0wg"
          />
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-1etrxmq"
    >
      <h2
        class="chakra-heading css-2zbbu2"
      >
        Login to Shop Bag
      </h2>
      <div
        class="chakra-stack css-886372"
        style="margin-top: 0px;"
      >
        <form>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <div
                class="chakra-input__left-element css-bqt2sq"
              >
                <svg
                  class="chakra-icon css-dw0gxi"
                  focusable="false"
                  viewBox="0 0 16 16"
                >
                  <path
                    d="M8 8C10.21 8 12 6.21 12 4C12 1.79 10.21 0 8 0C5.79 0 4 1.79 4 4C4 6.21 5.79 8 8 8ZM8 10C5.33 10 0 11.34 0 14V16H16V14C16 11.34 10.67 10 8 10Z"
                    fill="black"
                  />
                </svg>
              </div>
              <input
                class="chakra-input css-1tvziyz"
                data-testid="username"
                id="username"
                name="username"
                placeholder="Username"
                type="text"
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-ekfyzq"
            role="group"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <div
                class="chakra-input__left-element css-bqt2sq"
              >
                <svg
                  class="chakra-icon css-dw0gxi"
                  focusable="false"
                  viewBox="0 0 16 21"
                >
                  <path
                    d="M14 7H13V5C13 2.24 10.76 0 8 0C5.24 0 3 2.24 3 5V7H2C0.9 7 0 7.9 0 9V19C0 20.1 0.9 21 2 21H14C15.1 21 16 20.1 16 19V9C16 7.9 15.1 7 14 7ZM8 16C6.9 16 6 15.1 6 14C6 12.9 6.9 12 8 12C9.1 12 10 12.9 10 14C10 15.1 9.1 16 8 16ZM11.1 7H4.9V5C4.9 3.29 6.29 1.9 8 1.9C9.71 1.9 11.1 3.29 11.1 5V7Z"
                    fill="black"
                  />
                </svg>
              </div>
              <input
                aria-describedby="field-:r1:-helptext"
                class="chakra-input css-ncnfn5"
                data-testid="password"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
              />
              <div
                class="chakra-input__right-element css-18d7vni"
              >
                <button
                  aria-label="Call Sage"
                  class="chakra-button css-1eam5b1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-icon css-1d0iap9"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                      />
                      <path
                        d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="chakra-form__helper-text css-1r5werv"
              id="field-:r1:-helptext"
            />
            <button
              class="chakra-button css-lmjer0"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
        <div
          class="chakra-stack css-g6ykbf"
        >
          <p
            class="chakra-text css-1vl0whu"
          >
            Didn’t Have Account?
          </p>
          <a
            href="/auth/signin"
            style="margin-top: 0px; padding-left: 4px;"
          >
            <p
              class="chakra-text css-1mpbvs0"
            >
              Register
            </p>
          </a>
        </div>
        <div
          class="chakra-stack css-4okj21"
        >
          <div
            class="chakra-button__group css-wwgc5e"
            data-orientation="horizontal"
            role="group"
          >
            <button
              class="chakra-button css-17a8rn"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-13fvh6v"
                  focusable="false"
                  viewBox="0 0 16 16"
                >
                  <path
                    d="M14.537 6.69437H14V6.66671H7.99998V9.33337H11.7676C11.218 10.8857 9.74098 12 7.99998 12C5.79098 12 3.99998 10.209 3.99998 8.00004C3.99998 5.79104 5.79098 4.00004 7.99998 4.00004C9.01965 4.00004 9.94731 4.38471 10.6536 5.01304L12.5393 3.12737C11.3486 2.01771 9.75598 1.33337 7.99998 1.33337C4.31831 1.33337 1.33331 4.31837 1.33331 8.00004C1.33331 11.6817 4.31831 14.6667 7.99998 14.6667C11.6816 14.6667 14.6666 11.6817 14.6666 8.00004C14.6666 7.55304 14.6206 7.11671 14.537 6.69437Z"
                    fill="white"
                  />
                  <path
                    d="M2.10199 4.89704L4.29232 6.50337C4.88499 5.03604 6.32032 4.00004 7.99999 4.00004C9.01966 4.00004 9.94732 4.38471 10.6537 5.01304L12.5393 3.12737C11.3487 2.01771 9.75599 1.33337 7.99999 1.33337C5.43932 1.33337 3.21866 2.77904 2.10199 4.89704Z"
                    fill="white"
                  />
                  <path
                    d="M8 14.6667C9.722 14.6667 11.2867 14.0077 12.4697 12.936L10.4063 11.19C9.71452 11.7161 8.86915 12.0007 8 12C6.266 12 4.79367 10.8943 4.239 9.35132L2.065 11.0263C3.16834 13.1853 5.409 14.6667 8 14.6667Z"
                    fill="white"
                  />
                  <path
                    d="M14.537 6.69429H14V6.66663H8V9.33329H11.7677C11.5047 10.0721 11.0311 10.7177 10.4053 11.1903L10.4063 11.1896L12.4697 12.9356C12.3237 13.0683 14.6667 11.3333 14.6667 7.99996C14.6667 7.55296 14.6207 7.11663 14.537 6.69429Z"
                    fill="white"
                  />
                </svg>
              </span>
              <p
                class="chakra-text css-1gvoya1"
              >
                Google
              </p>
            </button>
            <button
              class="chakra-button css-1dtt86g"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-euhv5o"
                  focusable="false"
                  viewBox="0 0 8 16"
                >
                  <path
                    d="M2.04221 16V8.49234H0V5.78922H2.04221V3.4804C2.04221 1.66611 3.24856 0 6.02823 0C7.15368 0 7.98589 0.10488 7.98589 0.10488L7.92032 2.62914C7.92032 2.62914 7.07159 2.6211 6.14543 2.6211C5.14303 2.6211 4.98244 3.07014 4.98244 3.81544V5.78922H8L7.8687 8.49234H4.98244V16H2.04221Z"
                    fill="white"
                  />
                </svg>
              </span>
              <p
                class="chakra-text css-1gvoya1"
              >
                Facebook
              </p>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
